### ISSUE TITLE
new scipy release (1.7.0) breaks KNN tests

### ISSUE THREAD

**Issue by ksaur** [microsoft/hummingbird#527]
SciPy 1.7.0 was [released](https://github.com/scipy/scipy/releases/tag/v1.7.0) yesterday with changes to `cdist`.  It seems we need to remove an arg from our KNN converters, see [build](https://github.com/microsoft/hummingbird/runs/2877767604?check_suite_focus=true).

Ex:      `TypeError: cdist_old_weighted_minkowski_double_wrap() takes at most 5 arguments (6 given)`

for:
```
=========================== short test summary info ============================
FAILED tests/test_sklearn_kneighbors.py::TestSklearnKNeighbors::test_kneighbors_classifer_mahalanobis
FAILED tests/test_sklearn_kneighbors.py::TestSklearnKNeighbors::test_kneighbors_classifer_seuclidean
FAILED tests/test_sklearn_kneighbors.py::TestSklearnKNeighbors::test_kneighbors_classifer_wminkowski
FAILED tests/test_sklearn_kneighbors.py::TestSklearnKNeighbors::test_kneighbors_regressor_mahalanobis
FAILED tests/test_sklearn_kneighbors.py::TestSklearnKNeighbors::test_kneighbors_regressor_seuclidean
FAILED tests/test_sklearn_kneighbors.py::TestSklearnKNeighbors::test_kneighbors_regressor_wminkowski
===== 6 failed, 333 passed, 263 skipped, 95 warnings in 101.69s (0:01:41) ======
```

Troubleshooting in #526...I pinned just to verify that it's the new scipy version that breaks it, then we need to fix the args, unpin, and push.

We need to fix this before we merge #525.

**Comment by ksaur** [comment]
It seems that this is an SKL problem, and they are working on a fix [#20309](https://github.com/scikit-learn/scikit-learn/pull/20309).  (the torch converter is fine, it's creating the model in skl that is causing the test to fail.)

**Comment by ksaur** [comment]
Leaving this open until we can remove the pin in the ci/cd .yaml (#526)
